### PR TITLE
Use triggering file to determine weak vendors if when the test is run in a separate process

### PR DESCRIPTION
**Pull request by alexpott**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | I think so
| Fixed tickets | #25684
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
can you rebase on 3.4 please ?

**Comment by alexpott** [comment]
Rebased on 3.4. I can't work out how to write a test for this because atm all the weak vendor stuff is tested via phpt tests and a fake vendor.

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 1]
```diff
@@ -121,6 +121,7 @@ public static function register($mode = 0)
                     $msg = $parsedMsg['deprecation'];
                     $class = $parsedMsg['class'];
                     $method = $parsedMsg['method'];
+                    $isWeak = DeprecationErrorHandler::MODE_WEAK === $mode || (DeprecationErrorHandler::MODE_WEAK_VENDORS === $mode && $isVendor = $inVendors($parsedMsg['triggering_file']));
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: shouldn't $isVendor be initialized to false before this line?

Comment (nicolas-grekas) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: then L124: `$isWeak = $isWeak || ...`?

Comment (nicolas-grekas) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: self answering :)

> shouldn't $isVendor be initialized to false before this line?

not needing as logically this wouldn't make a different, but might still help the reader

> then L124: $isWeak = $isWeak || ...?

really not sure about that (might instead break the fix, you'll tell :) )

Comment (alexpott) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: Yeah that'd break the fix. The point is $isWeak is wrong at this point. Can always add a comment.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexpott.

**Comment by greg0ire** [comment]
Oh wow you fixed it, thanks! Could have pinged me though :P

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 2]
```diff
@@ -121,6 +121,12 @@ public static function register($mode = 0)
                     $msg = $parsedMsg['deprecation'];
                     $class = $parsedMsg['class'];
                     $method = $parsedMsg['method'];
+                    // If the deprecation has been triggered via
+                    // \Symfony\Bridge\PhpUnit\Legacy\SymfonyTestsListenerTrait::endTest()
+                    // then we need to use the serialized information to
+                    // determine if the error has been triggered from vendor
+                    // code.
+                    $isWeak = DeprecationErrorHandler::MODE_WEAK === $mode || (DeprecationErrorHandler::MODE_WEAK_VENDORS === $mode && $isVendor = $inVendors($parsedMsg['triggering_file']));
```
Comment (greg0ire) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php]: Very elegant solution :+1:

**Pull request closed without merging**